appholder/src/main/java/com/android/mdl/app/fragment/TransferDocumentFragment.kt

```diff
@@ -158,41 +158,37 @@ class TransferDocumentFragment : Fragment() {
         it.run()
     }
 
-    private fun sendResponseWithConsent() {
-        // Ask for biometric authentication when the device has biometric enrolled,
-        // otherwise just ask for consent in a dialog
-        if (BiometricManager.from(requireContext())
-                .canAuthenticate(BIOMETRIC_STRONG) == BIOMETRIC_SUCCESS
-        ) {
-            val promptInfo = BiometricPrompt.PromptInfo.Builder()
-                .setTitle(getString(R.string.bio_auth_title))
-                .setSubtitle(getSubtitle())
-                .setDescription(formatEntryNames(vm.getEntryNames()))
-                .setNegativeButtonText(getString(R.string.bio_auth_cancel))
-                .build()
-
-            val biometricPrompt = BiometricPrompt(this, executor, biometricAuthCallback)
-
-            val cryptoObject = vm.getCryptoObject()
-            // Displays the "log in" prompt.
-            if (cryptoObject != null) {
-                biometricPrompt.authenticate(promptInfo, cryptoObject)
+    private fun sendResponseWithConsent(forceLskf : Boolean) {
+        val promptInfoBuilder = BiometricPrompt.PromptInfo.Builder()
+            .setTitle(getString(R.string.bio_auth_title))
+            .setSubtitle(getSubtitle())
+            .setDescription(formatEntryNames(vm.getEntryNames()))
+
+        if (forceLskf) {
+            // TODO: this works only on Android 11 or later but for now this is fine
+            //   as this is just a reference/test app and this path is only hit if
+            //   the user actually presses the "Use PIN" button.  Longer term, we should
+            //   fall back to using KeyGuard which will work on all Android versions.
+            promptInfoBuilder.setAllowedAuthenticators(
+                BiometricManager.Authenticators.DEVICE_CREDENTIAL);
+        } else {
+            if (BiometricManager.from(requireContext()).canAuthenticate(BIOMETRIC_STRONG)
+                == BIOMETRIC_SUCCESS
+            ) {
+                promptInfoBuilder.setNegativeButtonText(getString(R.string.bio_auth_use_pin))
             } else {
-                biometricPrompt.authenticate(promptInfo)
+                // No biometrics enrolled, force use of KeyGuard
```

Comment:
This comment makes it sound like we're falling back to KeyGuard.

The above comment says "Longer term, we fall back to using KeyGuard". This seems contradictory to me. Line 180 indicates we're already using KeyGuard, so mission accomplished?

Reply:
Yeah, I meant to say "force use of LSKF".  Updated this.